src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy

```diff
@@ -644,95 +575,71 @@ public enum DiamondPearlPromos implements LogicCardInfo {
-          resistance F, MINUS20
-          pokeBody "Dragon Spirit", {
-            text "If Rayquaza is your Active Pok�mon and is damaged but not Knocked Out by an opponent�s attack, you may search your discard pile for an Energy card and attach it to Rayquaza ."
-            delayedA {
-            }
-          }
-          move "Final Blowup", {
-            text "200 damage. . Ignore this effect if you have no cards in your hand."
-            energyCost W, P, F, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-        };
+        return copy(SupremeVictors.RAYQUAZA_C_LV_X_146, this);
       case TROPICAL_WIND_DP48:
         return copy (TROPICAL_WIND_DP05, this)
-        /*basicTrainer (this) {
-					text "Flip a coin. If heads, remove 2 damage counters from each Active Pok�mon (remove 1 damage counter if a Pok�mon has only 1). If tails, each Active Pok�mon is now Asleep."
-					onPlay {
-					}
-					playRequirement{
-					}
-				}*/;
       case DIALGA_DP49:
         return basic (this, hp:HP100, type:METAL, retreatCost:2) {
           weakness R
           resistance P, MINUS20
           move "Time Call", {
-            text "Search your deck for a card that evolves from 1 of your Pok�mon and put it onto that Pok�mon. (This counts as evolving that Pok�mon.) Shuffle your deck afterward."
+            text "Search your deck for a card that evolves from 1 of your Pokémon and put it on that Pokémon. (This counts as evolving that Pokémon.) Shuffle your deck afterward."
             energyCost M
-            attackRequirement {}
-            onAttack {
-              damage 0
+            attackRequirement {
+              assert my.deck : "Deck is empty"
+            }
+            onAttack {
+              flip {
+                def names = my.all.collect {it.name }
+                def evolutionCard = deck.search ("Evolves from $names", {
+                  it.cardTypes.is(EVOLUTION) && names.contains(it.predecessor)
+                }).first()
+
+                if (evolutionCard) {
+                  def eligibleToEvolve = my.all.findAll({
+                    it.name == (evolutionCard as EvolutionPokemonCard).predecessor
+                  })
+                  def cardToEvolve = eligibleToEvolve.select("Evolve which one?")
+                  evolve(cardToEvolve, evolutionCard, OTHER)
+                }
+                shuffleDeck()
+              }
             }
           }
           move "Time Wager", {
-            text "100 damage. Flip 2 coins. If either of them is tails, this attack�s base damage is 50 instead of 100."
+            text "100 damage. Flip 2 coins. If either of them is tails, this attack's base damage is 50 instead of 100."
             energyCost M, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              def damageToDeal = 100
+
+              flip 2,{},{},[2:{
+                damageToDeal = 50
```

Comment:
This is backwards. It will deal 50 damage on HEADS HEADS and 100 damage otherwise.